runtime/js/src/main/kotlin/kotlinx/serialization/DynamicObjectParser.kt

```diff
@@ -56,6 +60,18 @@ class DynamicObjectParser(): AbstractSerialFormat() {
             }
         }
 
+        override fun decodeTaggedLong(tag: String): Long {
+            val obj = getByTag(tag)
+            val number = obj as? Double ?: throw SerializationException("$obj is not a Number")
+            val canBeConverted = number.isFinite() && floor(number) == number
+            if (!canBeConverted)
+                throw SerializationException("$number can't be represented as Long")
```

Comment:
Maybe "$number is not an integer"?

Reply:
I think it might confuse people with `kotlin.Int`?

Reply:
Maybe "$number is not finite, or has a non-zero fractional part"?